src/web3/onboard.ts

```diff
@@ -0,0 +1,80 @@
+import coinbaseModule from '@web3-onboard/coinbase'
+import gnosisModule from '@web3-onboard/gnosis'
+import injectedModule from '@web3-onboard/injected-wallets'
+import ledgerModule from '@web3-onboard/ledger'
+import { init } from '@web3-onboard/react'
+import trezorModule from '@web3-onboard/trezor'
+import walletConnectModule from '@web3-onboard/walletconnect'
+
+import { AelinIconString, AelinLogoString } from '../components/assets/Logo'
+import { Chains, chainsConfig } from '@/src/constants/chains'
+
+const EMAIL_CONTACT = process.env.NEXT_PUBLIC_EMAIL_CONTACT as string
+const APP_URL = process.env.NEXT_PUBLIC_APP_URL as string
+const APP_NAME = process.env.NEXT_PUBLIC_APP_NAME as string
+
+const injected = injectedModule()
+const gnosis = gnosisModule()
+const walletConnect = walletConnectModule()
+const coinbase = coinbaseModule()
+const ledger = ledgerModule()
+const trezor = trezorModule({
+  email: EMAIL_CONTACT,
+  appUrl: APP_URL,
+})
+
+init({
+  wallets: [injected, coinbase, walletConnect, gnosis, trezor, ledger],
+  chains: [
+    {
+      id: chainsConfig[Chains.mainnet].id,
```

Comment:
Based on the [docs](https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/core/README.md#options), the chain id should be hex encoded.

Some weeks ago, I added this value to the `chainsConfig` object. It now has a `chainIdHex` field, so you can use this on each network.